packages/webapp/src/components/Modals/WaterUseCalculatorModal/index.jsx

```diff
@@ -1,32 +1,315 @@
+  const { t } = useTranslation();
+  const { register, getValues, watch, control, setValue } = formState();
+  const { estimated_flow_rate, estimated_irrigation_duration } = getValues();
+
+  useEffect(() => {
+    if (estimated_flow_rate && estimated_irrigation_duration) {
+      setTotalWaterUsage(() => estimated_flow_rate * estimated_irrigation_duration);
+    }
+  }, [estimated_irrigation_duration, estimated_flow_rate]);
+
+  const FLOW_RATE = 'estimated_flow_rate';
+  const FLOW_RATE_UNIT = 'estimated_flow_rate_unit';
+  const DEFAULT_LOCATION_FLOW_RATE = 'default_location_flow_rate';
+  const ESTIMATED_DURATION = 'estimated_irrigation_duration';
+  const ESTIMATED_DURATION_UNIT = 'estimated_irrigation_duration_unit';
+
+  return (
+    <>
+      <Unit
+        register={register}
+        displayUnitName={FLOW_RATE_UNIT}
+        label={t('ADD_TASK.IRRIGATION_VIEW.ESTIMATED_FLOW_RATE')}
+        hookFormSetValue={setValue}
+        hookFormGetValue={getValues}
+        hookFromWatch={watch}
+        name={FLOW_RATE}
+        unitType={water_valve_flow_rate}
+        max={999999.99}
+        system={system}
+        control={control}
+        onChangeUnitOption={(e) => {
+          e.label === 'l/h' && setTotalWaterUsage(() => totalWaterUsage / 60);
+          e.label === 'l/m' && setTotalWaterUsage(() => totalWaterUsage * 60);
+        }}
+      />
+
+      <Checkbox
+        label={t('ADD_TASK.IRRIGATION_VIEW.DEFAULT_LOCATION_FLOW_RATE')}
+        sm
+        style={{ marginTop: '10px', marginBottom: '30px' }}
+        hookFormRegister={register(DEFAULT_LOCATION_FLOW_RATE)}
+      />
+
+      <Unit
+        register={register}
+        displayUnitName={ESTIMATED_DURATION_UNIT}
+        label={t('ADD_TASK.IRRIGATION_VIEW.ESTIMATED_DURATION')}
+        hookFormSetValue={setValue}
+        hookFormGetValue={getValues}
+        hookFromWatch={watch}
+        name={ESTIMATED_DURATION}
+        unitType={estimated_duration}
+        max={999999.99}
+        system={system}
+        control={control}
+        style={{ paddingBottom: '32px' }}
+        onChangeUnitOption={(e) => {
+          e.label === 'h' && setTotalWaterUsage(() => totalWaterUsage / 60);
+          e.label === 'm' && setTotalWaterUsage(() => totalWaterUsage * 60);
+        }}
+      />
+
+      <TotalWaterUsage totalWaterUsage={totalWaterUsage} />
+    </>
+  );
+};
+
+const WaterUseDepthCalculator = ({ system, setTotalWaterUsage, totalWaterUsage, formState }) => {
+  const { t } = useTranslation();
+  const { register, getValues, watch, control, setValue } = formState();
+  const { irrigated_area, application_depth } = getValues();
+  const modalState = useForm({ mode: 'onChange', shouldUnregister: false });
+  const { percentage_location_irrigated } = modalState.getValues();
+
+  const locationSize = useSelector(cropLocationsSelector).filter(
+    (location) => location.location_id === getValues().locations[0].location_id,
```

Comment:
can use optional for null safety e.g. location?.location_id === getValues()?.locations[0]?.location_id